r/configure

```diff
@@ -74,18 +75,29 @@ else
   fi
 
   if [ "$PKGCONFIG_CFLAGS" ] || [ "$PKGCONFIG_LIBS" ]; then
-    echo "Arrow C++ libraries found via pkg-config"
+    echo "*** Arrow C++ libraries found via pkg-config"
     PKG_CFLAGS="$PKGCONFIG_CFLAGS"
     PKG_LIBS=${PKGCONFIG_LIBS}
+
+    # Check for version mismatch
+    PC_LIB_VERSION=`pkg-config --modversion arrow`
+    echo $PC_LIB_VERSION | grep -e 'SNAPSHOT$' >/dev/null 2>&1
```

Comment:
Is this a debug print?
Should we remove this?

Reply:
Unless I'm mistaken, it doesn't print, it redirects all output. The point is whether the `grep` exits 0 or 1 based on whether or not it matches. 

Maybe you know a better POSIX shell compliant way to test whether the pkg-config version contains "SNAPSHOT"?

Reply:
Ah, I understand. Sorry.

How about `if pkg-config "arrow != ${VERSION}"; then` instead of `if [ $? -eq 1 ] && ...`?

R never use `X.Y.Z-SNAPSHOT` version. So it always `false` for `X.Y.Z-SNAPSHOT` C++.

Reply:
Ah, `pkg-config --exists "arrow != ${VERSION}"` is better than no `--exists`.
Because it describes what we want to do.

Reply:
So, I want to warn if C++ version != R version, except I don't need to warn if C++ version is a SNAPSHOT version, because if I'm developing locally the versions won't match (in master, C++ is 1.0.0-SNAPSHOT while R is 0.16.0.9000).

Reply:
Sorry... I misunderstood...

We need to use `grep` for it as you wrote.

If I write this logic, I'll write `grep` in `if`:

```shell
if ! echo "${PC_LIB_VERSION}" | grep -q -e 'SNAPSHOT$' && [ "${PC_LIB_VERSION}" != "${VERSION}" ]; then
```